packages/core/src/client/validate.ts

```diff
@@ -103,6 +103,68 @@ export const validateQuery = async (
   }
 };
 
+/**
+ * Find all table names in a SQL query.
+ *
+ * @param sql - SQL query
+ */
+export const findTableNames = async (sql: string) => {
+  // @ts-ignore
+  const Parser = await import(/* webpackIgnore: true */ "pg-query-emscripten");
+
+  if (sql.length > 5_000) {
+    throw new Error("Invalid query");
+  }
+
+  const { parse } = await Parser.default();
+  const parseResult = parse(sql) as {
```

Comment:
Perhaps this function should accept a `parseResult` - I think at the moment we'll parse the query twice needlessly

Reply:
Added a LRU cache for `findTableNames` that will give us the desired behavior.